ButtonFunctions/__init__.py

```diff
@@ -16,15 +18,26 @@ def supress(application):
 
 def save_curve(application) -> None:
     """Save Curve.
-
-    Saves a curve into a csv format.
+    Saves the retrieved data of each of the sensors into csv files.
+    Also generates an image of the graph of Data objects with a designated plot.
+    All the files are stored in a folder with name indicating timestamp.
     """
-    file_path = 'data.csv'
-    with open(file_path, 'w') as myfile:
-        wr = csv.writer(myfile, quoting=csv.QUOTE_ALL)
-        wr.writerow(application.times)
-        wr.writerow(application.forces)
+    timestr = time.strftime("%Y-%m-%d--%H-%M-%S")
+    dir_path = os.path.join(os.path.dirname(os.path.dirname(__file__)), 'Saves', 'Save--'+timestr)
+    os.makedirs(dir_path)
 
+    for sensor in application.sensor_data:
+        file_path = os.path.join(dir_path, sensor.name)
+        with open(file_path+'.csv', 'w', newline ='') as myfile:
```

Comment:
it's preferred ; as the character separator. Perhaps it's a good idea to change newline='' to newline=';'

Reply:
The newline argument for the `open` function controls how universal newlines work, it's inputs can be `None`, `''` , `'\n'`, `'\r'`, and `'\r\n'`. Perhaps you're mistaking it with the `delimiter` arg sometimes used in numpy functions. Since we are using a csv writer, it already standardizes to a comma `,` for separating values.